spanner/client.go

```diff
@@ -438,10 +438,11 @@ func (c *Client) BatchReadOnlyTransaction(ctx context.Context, tb TimestampBound
 
 	t := &BatchReadOnlyTransaction{
 		ReadOnlyTransaction: ReadOnlyTransaction{
-			tx:              tx,
-			txReadyOrClosed: make(chan struct{}),
-			state:           txActive,
-			rts:             rts,
+			tx:                       tx,
+			txReadyOrClosed:          make(chan struct{}),
+			state:                    txActive,
+			rts:                      rts,
+			isLongRunningTransaction: true,
```

Comment:
Yes, 'BatchReadOnlyTransaction' creates its own session and does not fetch session from pool.
https://github.com/googleapis/google-cloud-go/blob/f609b3cf831fb89c45386f81d0047560120cb3f4/spanner/client.go#L416-L423

1. When I wrote this I was wondering how is retry handled for `BatchReadOnlyTransaction`, if retry was there is that new session being used from pool or a new session is created and as I was not clear at that time, I used this to prevent something unexpected happening.
But looks like `BatchReadOnlyTransaction` does not have any retry logic in `Go` client library. Is this the same with `Java` also?
2. Also one more confusion was what should happen if a 'BatchReadOnlyTransaction' executes a normal query. Example https://github.com/googleapis/google-cloud-go/blob/f609b3cf831fb89c45386f81d0047560120cb3f4/spanner/integration_test.go#L3326-L3335
Logically in this case the query is not eligible for long running though the transaction can be long running.

I think marking `isLongRunningTransaction ` as `true` or `false` for `BatchReadOnlyTransaction` does not have any effect.

Reply:
Yes, agree with Harsha here BatchReadOnlyTransaction are not retried and create their own session, so setting isLongRunningTransaction does not have any impact